daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/FatContractInstance.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+import data.{Bytes, Ref, Time}
+import value.{CidContainer, Value}
+
+// This should replace value.ContractInstance in the whole daml/canton codespace
+// TODO: Rename to ContractInstance once value.ContractInstance is properly deprecated
+sealed abstract class FatContractInstance extends CidContainer[FatContractInstance] {
+  val version: TransactionVersion
+  val contractId: Value.ContractId
+  val templateId: Ref.TypeConName
+  val createArg: Value
+  val contractKey: Option[GlobalKey]
+  val maintainers: Set[Ref.Party]
+  val nonMaintainerSignatories: Set[Ref.Party]
+  val nonSignatoryStakeholders: Set[Ref.Party]
+  val createTime: Time.Timestamp
+  val cantonData: Bytes
+  private[lf] def toImplementation: FatContractInstanceImpl =
+    this.asInstanceOf[FatContractInstanceImpl]
+  require(maintainers.isEmpty || contractKey.isDefined)
+  final lazy val signatories: Set[Ref.Party] = maintainers | nonMaintainerSignatories
+  final lazy val stakeholders: Set[Ref.Party] = signatories | nonSignatoryStakeholders
+  def updateCreateTime(updatedTime: Time.Timestamp): FatContractInstance
```

Comment:
this is the ledger-effective time isn't it? consider using a name that leaves space for the record time (i.e., the sequencing time of the tx); e.g.,
```
effectiveAt
recordedAt
```

Reply:
it's not clear to me record time actually makes sense here. record time is in the context of a domain. So you would also need to include the domain id for that to make sense. But then why should anyone care which domain a contract was created on? You might not even be connected to that domain.

But changing the names to make it clear which one it is does make sense.

Reply:
Ah good point. Yes. Given that contract instances can be reassigned record-time does not make sense here.